src/Build/Graph/ProjectGraph.cs

```diff
@@ -51,8 +65,34 @@ public ProjectGraph(string entryProjectFile)
             Dictionary<string, string> globalProperties,
             string toolsVersion)
         {
-            LoadGraph(entryProjectFile, projectCollection, globalProperties, toolsVersion);
-            EntryProjectNode = _allParsedProjects[entryProjectFile];
+            ErrorUtilities.VerifyThrowArgumentNull(projectCollection, nameof(projectCollection));
+
+            PropertyDictionary<ProjectPropertyInstance> globalPropertyDictionary;
+            if (globalProperties == null)
+            {
+                globalPropertyDictionary = new PropertyDictionary<ProjectPropertyInstance>(0);
+            }
+            else
+            {
+                globalPropertyDictionary = new PropertyDictionary<ProjectPropertyInstance>(globalProperties.Count);
+                foreach (KeyValuePair<string, string> entry in globalProperties)
+                {
+                    globalPropertyDictionary[entry.Key] = ProjectPropertyInstance.Create(entry.Key, entry.Value);
+                }
+            }
+
+            if (string.IsNullOrEmpty(toolsVersion))
+            {
+                // TODO: This should actually be the project's tools version, but we haven't actually loaded it yet.
```

Comment:
We should perhaps discuss what we actually want to happen with this tools version thing all up before trying to solve this.

To solve this properly, we'll need to load the XML pretty early instead of passing around just the file name.